test/test_sphere.py

```diff
@@ -22,7 +22,7 @@ def test_diameter_various_spheres_returns_correct_diameter(self):
         ]
         expecteds = [0, 2, 0, 20.2]
         for (case, expect) in zip(cases, expecteds):
-            with self.subTest(case=case, expect=expect):
```

Comment:
May be worth keeping the args around in our own tests though. Distinguishing subtests is useful.

In general, revisiting subtests later to introduce the optional kwargs seems nice to have.

Reply:
Grand idea bro.